### ISSUE TITLE
USER ISSUE: Friend cannot connect

### ISSUE THREAD

**Issue by Nimarcs** [StrangeLoopGames/EcoIssues#14698]
**Version:** 0.8.3.3 beta

**Steps to Reproduce:**
1. Host a new world 
2. ask a friend to try to connect

**Expected behavior:**
The friend should join and will would play in happiness

**Actual behavior:**
launch an infinite loading screen with no text

**Do you have mods installed? Does issue happen when no mods are installed?:**
launch an infinite loading screen with no text

**Comment by beastyn** [comment]
1.Does your friend have this problem only with your server?
Can a friend try to connect to official servers (Sea Otter and Giant Tortoise) and provide log files?
Instruction is here https://github.com/StrangeLoopGames/EcoIssues/wiki/Getting-the-Eco-game-log-file
2. Can a friend see servers in the list normally? With ping and server info?
3. Do you and your friend have Firewall or Antivirus enabled?
4. Which kind of internet connection are you using and what about a friend?
5. Please provide your PC specification (include Operating System) for you and your friend.
6. Did your friend meet this issue after updating to the "latest build" or had it before?
7. Do you have external IP address and did you use it in config file?

**Comment by Nimarcs** [comment]
In fact i think the problem come from me,
I can connect to server but not all and i cannot always see all the server sometimes only recent one appear but in offline (appart from The Flying pineapple ¯\_(ツ)_/¯) and with luck and refresh i see the official one and can connect to sea otter or giant panda (and i've never seen giant tortoise)
![image](https://user-images.githubusercontent.com/25551779/74316271-853d7380-4d79-11ea-9d29-79fc6f746e2a.png)
![image](https://user-images.githubusercontent.com/25551779/74316313-9be3ca80-4d79-11ea-8625-7ee86b4372c3.png)
[Player.log](https://github.com/StrangeLoopGames/EcoIssues/files/4190661/Player.log)
[Player-prev.log](https://github.com/StrangeLoopGames/EcoIssues/files/4190662/Player-prev.log)



I only have windows defender and we are using wifi but are not in LAN

Here is my pc spec
Nom du système d’exploitation	Microsoft Windows 10 Famille
Version	10.0.18362 Numéro 18362
Autre description du système d’exploitation	Non disponible
Fabricant du système d’exploitation	Microsoft Corporation
Ordinateur	DESKTOP-2HKI3BM
Fabricant	System manufacturer
Modèle	System Product Name


Type	PC à base de x64
Référence (SKU) du système	SKU
Processeur	AMD Ryzen 5 1500X Quad-Core Processor, 3500 MHz, 4 cœur(s), 8 processeur(s) logique(s)
Version du BIOS/Date	American Megatrends Inc. 0515, 30/03/2017
Version SMBIOS	3.0
Version du contrôleur embarqué	255.255
Mode BIOS	UEFI
Fabricant de la carte de base	ASUSTeK COMPUTER INC.
Produit de la carte de base	PRIME B350-PLUS
Version de la carte de base	Rev X.0x
Rôle de la plateforme	Bureau
État du démarrage sécurisé	Désactivé
Configuration de PCR 7	Liaison impossible
Répertoire Windows	C:\Windows
Répertoire système	C:\Windows\system32
Périphérique de démarrage	\Device\HarddiskVolume2
Option régionale	France
Couche d’abstraction matérielle	Version = "10.0.18362.387"
Utilisateur	DESKTOP-2HKI3BM\Marcus
Fuseaux horaires	Paris, Madrid
Mémoire physique (RAM) installée	8,00 Go
Mémoire physique totale	7,93 Go
Mémoire physique disponible	2,94 Go
Mémoire virtuelle totale	18,4 Go
Mémoire virtuelle disponible	6,59 Go
Espace pour le fichier d’échange	10,5 Go
Fichier d’échange	C:\pagefile.sys
Protection DMA du noyau	Désactivé
Sécurité basée sur la virtualisation	Désactivé
Prise en charge du chiffrement d’appareil%s	Motifs de l’échec du chiffrement de l’appareil automatique: TPM inutilisable, Liaison PCR7 non prise en charge, L'interface du test de sécurité du matériel a échoué et l'appareil n'est pas en veille moderne., Détection de bus/d’appareils compatibles DMA non autorisés, TPM inutilisable
Hyper-V - Extensions du mode de moniteur des ordinateurs virtuels	Oui
Hyper-V - Extensions de la conversion des adresses de second niveau	Oui
Hyper-V - Virtualisation activée dans le microprogramme	Non
Hyper-V - Protection de l’exécution des données	Oui
https://www.userbenchmark.com/UserRun/24575918 (not launched with only the benchmark)

And my friend config:
Intel Core i5 9600KF
Be Quiet Pure Rock Slim
MSI GeForce GTX 1660 SUPER Aero ITX OC 
MSI B360-A PRO 
HyperX Predator DDR4 1 x 16 Go 2666 MHz CAS 13 
Samsung Serie 860 EVO 500 Go 
Asus Graveur DVD - DRW-24D5MT - Noir - OEM 
Cooler Master N300 
Gigabyte B700H
He has Windows 10 and i don't know if he has a antivirus

We both just downloaded the game and we have tried reinstall it and restart it nothing work
I never modified (or seen) a configfile
We tried to create a world and connect to play with steam
We tried to exange role but he cannot connect but has an error message "timed out" not me i'm just in infinite loding screen.

**Comment by D3nnis3n** [comment]
Do you have a IPV4 address?

**Comment by Nimarcs** [comment]
Yes i have one,
Big update the problem has been fixed, i needed to create an Network.eco in my game file (in D:\Program Files (x86)\Steam\steamapps\common\Eco\Eco_Data\Server\Configs) and change  "MaxConnections": 3.
Now my friend can connect. 
The fact that I can't connect to server has been fixed after I unistalled Steam version, installed the Zip version (available on your Website) then reinstalled Steam version but i don't know why.
PS (i have reinstalled the steam version 6 time in total and the last one (after the zip version) was the only one to work so i suppose that was the solution.
If you have other question to make sure this bug don't happen to other ask me !
